### PR TITLE
More options for serialization/deserialization functions

### PR DESCRIPTION
**Pull request by DanieleDiBenedetto**

### CODE REVIEW

File: include/zendoo_mc.h [hunk 1]
```diff
@@ -559,22 +573,25 @@ extern "C" {
...
+     * Deserialize a proof, in compressed or uncompressed form, depending on the value of `compressed` flag,
+     * from `sc_proof_bytes` and return an opaque pointer to it.
      * If `semantic_checks` flag is set, semantic checks on the proof will be performed.
      */
     sc_proof_t* zendoo_deserialize_sc_proof(
         const BufferWithSize* sc_proof_bytes,
         bool semantic_checks,
-        CctpErrorCode* ret_code
+        CctpErrorCode* ret_code,
+        bool compressed = true
```
Comment (ptagl) [inline include/zendoo_mc.h:594]: As a further improvement, could we get rid of the `compressed` flag when deserializing? Can it be deduced from the serialized bytes themselves?

File: include/zendoo_mc.h [hunk 2]
```diff
@@ -928,12 +1005,16 @@ extern "C" {
         uint32_t num_constraints,
         const path_char_t* params_dir,
         size_t params_dir_len,
-        CctpErrorCode* ret_code
+        CctpErrorCode* ret_code,
+        bool compress_vk = true,
+        bool compress_pk = true
```
Comment (ptagl) [inline include/zendoo_mc.h:1010]: Is there any case in which we should compress the verification key and keep the proving key uncompressed (or the other way around)? If not, we might use a unique `compress` flag for both keys.

Comment (DanieleDiBenedetto) [reply include/zendoo_mc.h:1010]: In reality we never want to save the pk uncompressed (cause it's very big). The vk is pretty small so it's doable instead

File: include/zendoo_mc.h (hunk 1 above)
Comment (DanieleDiBenedetto) [reply include/zendoo_mc.h:594]: Unfortunately I don't think so. We would need to know a priori the exact compressed and uncompressed size of the particular instance, and this is not possible (at least not for every type)

Comment (ptagl) [reply include/zendoo_mc.h:594]: Isn't it possible to use an approach similar to `zendoo_get_sc_proof_proving_system_type_from_file()`?

Comment (DanieleDiBenedetto) [reply include/zendoo_mc.h:594]: The proof, pk and vk size is not fixed and can change, regardless of the type

Comment (ptagl) [reply include/zendoo_mc.h:594]: Then, what about prepending a byte to the serialized file to specify if it is compressed or not (immediately after or before the _type_ byte)?

**Review by alsala — APPROVED** [review]

**Review by ptagl — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

**Pull request merged**